src/pages/Portfolio/LendingStats/LendingStatsPage.vue

```diff
@@ -157,19 +159,46 @@ export default {
 	},
 	created() {
 		this.apollo.query({
-			query: userAchievementProgressQuery,
+			query: lenderProfileBadgeDataQuery,
+			variables: {
+				contentType: 'challenge',
+				limit: 200,
+			}
 		}).then(({ data }) => {
-			this.allAchievements = data?.userAchievementProgress?.lendingAchievements ?? [];
+			this.allAchievements = data ?? {};
 		});
 	},
 	computed: {
 		completedAchievements() {
-			return this.allAchievements.filter(
-				achievement => achievement.milestoneProgress?.[0]?.milestoneStatus === 'COMPLETE'
-			);
+			const completedBadgesArr = this.getBadgesData();
+
+			completedBadgesArr.sort((a, b) => {
+				return new Date(a.earnedAtDate) - new Date(b.earnedAtDate);
+			});
+
+			return completedBadgesArr;
 		},
+		badgesLoading() {
+			return !this.badgesData.length;
```

Comment:
What happens if the user is new and has no badges?

Reply:
Ah, so it's the data for the badges themselves and will always contain data for users?